src/components/network/network-table.js

```diff
@@ -75,40 +204,125 @@ const NetworkTable = (props) => {
     const allDisplayedColumnsNames = useSelector(
         (state) => state.allDisplayedColumnsNames
     );
+    const allLockedColumnsNames = useSelector(
+        (state) => state.allLockedColumnsNames
+    );
     const fluxConvention = useSelector((state) => state[PARAM_FLUX_CONVENTION]);
 
+    const [lineEdit, setLineEdit] = useState(undefined);
     const [popupSelectColumnNames, setPopupSelectColumnNames] = useState(false);
     const [rowFilter, setRowFilter] = useState(undefined);
+    const [columnSort, setColumnSort] = useState(undefined);
     const [tabIndex, setTabIndex] = useState(0);
     const [selectedColumnsNames, setSelectedColumnsNames] = useState(new Set());
+    const [lockedColumnsNames, setLockedColumnsNames] = useState(new Set());
     const [scrollToIndex, setScrollToIndex] = useState(-1);
     const [manualTabSwitch, setManualTabSwitch] = useState(true);
     const [selectedDataKey, setSelectedDataKey] = useState(new Set());
 
+    const isLineOnEditMode = useCallback(
+        (rowData) => {
+            return lineEdit && rowData.id === lineEdit.id;
+        },
+        [lineEdit]
+    );
+
     const intl = useIntl();
 
     const intlRef = useIntlRef();
 
     useEffect(() => {
+        const allDisplayedTemp = allDisplayedColumnsNames[tabIndex];
         setSelectedColumnsNames(
-            new Set(JSON.parse(allDisplayedColumnsNames[tabIndex]))
+            new Set(allDisplayedTemp ? JSON.parse(allDisplayedTemp) : [])
         );
+        setLineEdit({});
     }, [tabIndex, allDisplayedColumnsNames]);
 
+    useEffect(() => {
+        const allLockedTemp = allLockedColumnsNames[tabIndex];
+        setLockedColumnsNames(
+            new Set(allLockedTemp ? JSON.parse(allLockedTemp) : [])
+        );
+    }, [tabIndex, allLockedColumnsNames]);
+
     useEffect(() => {
         const resource = TABLES_DEFINITION_INDEXES.get(tabIndex).resource;
         if (!props.network) return;
         props.network.useEquipment(resource);
     }, [props.network, tabIndex]);
 
+    const filter = useCallback(
+        (cell) => {
+            if (!rowFilter) return true;
+            return (
+                [...selectedDataKey].find((key) =>
+                    rowFilter.test(cell[key])
+                ) !== undefined
+            );
+        },
+        [rowFilter, selectedDataKey]
+    );
+
+    const formatCell = useCallback(
+        (rowData, columnDefinition) => {
+            let value = rowData[columnDefinition.dataKey];
+            if (columnDefinition.cellDataGetter) {
+                value = columnDefinition.cellDataGetter(rowData, props.network);
```

Comment:
keep the same API cellDataGetter as the previous one ?

Reply:
Can you elaborate ? I do not understand.


By "previous one", you mean the old definition in config-tables.js ? Before modifying it, I scoured the code and checked with @niconoir to make sure there would not be negative side effects.


Or maybe you mean that we could use an else to not execute line 269 if columnDefinition.cellDataGetter is defined ?